index.php

```diff
@@ -20,10 +20,46 @@
 if (isset($_GET['breed'])) {
     $cats = $catHydrator->createCatEntitiesArray((int)$_GET['breed']);
     foreach ($cats as $cat) {
-        $catsHtml .= '<div class="cat-image"><img src="' . $cat->getImage() . '" alt="' . $breeds[$cat->getBreed() - 1]['breed'] . '"></div>';
+        $catsHtml .= '<div class="cat-image">
+        <div class="favorite-icon-container">
+        <form action="index.php" method="post">
+        <input class="cat-id-input" name="breedID" value="' . $cat->getBreed() . '">
+        <input class="cat-id-input" name="newFavourite" value="' . $cat->getID() . '">
+        <img class="favorite-icon" src="';
+
+        $breedWanted = $_GET['breed'] - 1;
+
+        if($cat->getID() == $breeds[$breedWanted]['favourite_id']) {
+            $catsHtml .= 'images/fav-icon-full.svg';
+            } else { $catsHtml .= 'images/fav-icon-empty.svg';
+        }
+
+        $catsHtml .=  '" alt="">
+        </form>
+        </div>
+        <img src="' . $cat->getImage() . '" alt="A cat">
+        </div>';
     }
 }
 
+$favouriteResponseDisplay = '';
+
+//Checks if a POST is set for newFavourite and breedID.
+//Then updates row in Database setting the breed row to have that newFavourite ID.
+if (isset($_POST['newFavourite']) && isset($_POST['breedID'])) {
+    $favouriteSql = $dbConnection->prepare('UPDATE `breed` SET favourite_id = :newFavourite WHERE id= :breedID;');
+    //newFavourite is the ID of the image of the favourite cat.
+    $favouriteSql->bindParam('newFavourite', $_POST['newFavourite'], PDO::PARAM_INT);
+    //breedID is the ID of the breed of cat.
+    $favouriteSql->bindParam('breedID', $_POST['breedID'], PDO::PARAM_INT);
+    $favouriteSql->execute();
+    $favouriteResponseMessage = 'Cat successfully favourited';
+    $favouriteResponseDisplay .= '<div class="favorite-response-message">' . $favouriteResponseMessage . '</div>';
```

Comment:
`$favouriteResponseDisplay .= '<div class="favorite-response-message">' . $favouriteResponseMessage . '</div>';` should be after the else, otherwise you're not putting it in the div.